### PR TITLE
GP-965 integrate `openedx-completion-aggregator`

### PR DESCRIPTION
**Pull request by qasimgulzar**

### CODE REVIEW

File: openedx/features/genplus_features/genplus_learning/signals.py [hunk 1]
```diff
@@ -147,3 +133,8 @@ def program_deleted(sender, instance, **kwargs):
 def program_updated(sender, instance, created, **kwargs):
     if not created:
         ProgramCache.clear_mapping_for_all_courses(instance)
+@receiver(SignalHandler.course_published)
+def course_published(course_key, **kwargs):
+    print(kwargs)
```
Comment (ziafazal) [inline openedx/features/genplus_features/genplus_learning/signals.py]: remove print statements. what we need this `course_published` handler?

File: openedx/features/genplus_features/genplus_learning/api/v1/serializers.py [hunk 2]
```diff
@@ -156,7 +155,7 @@ def get_units(self, obj):
 
                 units_context[unit.pk] = {
                     'is_locked': is_locked,
-                    'progress': completion.progress if completion else 0,
+                    'progress': get_aggregated_progress(request._request, course_key),
```
Comment (ziafazal) [inline openedx/features/genplus_features/genplus_learning/api/v1/serializers.py:158]: I think we should not use private variable(`_request`) here. What if we pass `request` as it is?

**Review by ziafazal — COMMENTED** [review]
@qasimgulzar overall looks good I have couple of questions.

File: openedx/features/genplus_features/genplus_learning/api/v1/serializers.py (hunk 2 above)
Comment (qasimgulzar) [reply openedx/features/genplus_features/genplus_learning/api/v1/serializers.py:158]: It prompts exception, "The `request` argument must be an instance of `django.http.HttpRequest`, not `rest_framework.request.Request`" when calling view directly
image

File: openedx/features/genplus_features/genplus_learning/signals.py (hunk 1 above)
Comment (qasimgulzar) [reply openedx/features/genplus_features/genplus_learning/signals.py]: I removed it already, this signal was for testing it is no more there

**Review by shavaiz-hussain — APPROVED** [review]

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**